test/test_data/test_feature_validator.py

```diff
@@ -417,10 +416,10 @@ def test_list_to_dataframe(openml_id):
         if is_numeric_dtype(X_pandas[col].dtype):
             # convert dtype translates 72.0 to 72. Be robust against this!
             assert is_numeric_dtype(transformed_X[i].dtype)
-        else:
-            if X_pandas[col].dtype.name == "category":
-                if transformed_X[i].dtype.name != "string" and transformed_X[i].dtype.name != "category":
-                    assert False
+        elif is_string_dtype(X_pandas[col].dtype):
+            assert is_string_dtype(X_pandas[col].dtype)
+        elif is_categorical_dtype(X_pandas[col].dtype):
+            assert is_categorical_dtype(X_pandas[col].dtype)
```

Comment:
Coud you please add a final else that raises a NotImplementedError (just in case something else happens we get an explicit error)?